bootstrap/api/v1alpha1/agentbootstrapconfig_types.go

```diff
@@ -87,6 +87,7 @@ type AgentBootstrapConfigStatus struct {
 	// AgentRef references the agent this agent bootstrap config has booted
 	AgentRef *corev1.LocalObjectReference `json:"agentRef,omitempty"`
 
+	// TODO: should this be on the spec?
```

Comment:
I believe we initially moved it here to prevent the user from setting this field. 
What are the reasons to move it to spec?

Reply:
It's desired state. We want the URL to be used in the bootstrap process. And of course in a declarative API, we "spec"ify our desired state in the `spec`. :)

The status should be reserved for observations about the world that were taken the last time the owning controller reconciled this resource.

Reply:
Ah I see! That makes sense, thank you for the explanation! CC @rccrdpccl